src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -671,49 +727,88 @@ GenericParameterValue genericParam
-								AddReturnValue (MultiValueLattice.Top);
-								break;
-							case SystemTypeValue systemTypeValue:
-								AddReturnValue (new NullableSystemTypeValue (typeValue.RepresentedType, new SystemTypeValue (systemTypeValue.RepresentedType)));
-								break;
-							// Generic Parameters and method parameters with annotations
-							case ValueWithDynamicallyAccessedMembers damValue:
-								AddReturnValue (new NullableValueWithDynamicallyAccessedMembers (typeValue.RepresentedType, damValue));
-								break;
-							// Everything else assume it has no annotations
-							default:
-								AddReturnValue (GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes));
-								break;
+
+						// Nullables without a type argument are considered SystemTypeValues
+						if (typeValue.RepresentedType.IsTypeOf ("System", "Nullable`1")) {
+							foreach (var argumentValue in argumentValues[0]) {
+								if ((argumentValue as ArrayValue)?.TryGetValueByIndex (0, out var underlyingMultiValue) == true) {
+									foreach (var underlyingValue in underlyingMultiValue) {
+										switch (underlyingValue) {
+										// Don't warn on these types - it will throw instead
+										case NullableValueWithDynamicallyAccessedMembers:
+										case NullableSystemTypeValue:
+										case SystemTypeValue maybeArrayValue when maybeArrayValue.RepresentedType.IsTypeOf ("System", "Array"):
+											AddReturnValue (MultiValueLattice.Top);
+											break;
+										case SystemTypeValue systemTypeValue:
+											AddReturnValue (new NullableSystemTypeValue (typeValue.RepresentedType, new SystemTypeValue (systemTypeValue.RepresentedType)));
+											break;
+										// Generic Parameters and method parameters with annotations
+										case ValueWithDynamicallyAccessedMembers damValue:
+											AddReturnValue (new NullableValueWithDynamicallyAccessedMembers (typeValue.RepresentedType, damValue));
+											break;
+										// Everything else assume it has no annotations
+										default:
+											// This returns just Nullable<> SystemTypeValue - so some things will work, but GetUnderlyingType won't propagate anything
+											// It's special cased to do that.
+											AddReturnValue (value);
+											break;
+										}
+									}
+								} else {
+									// This returns just Nullable<> SystemTypeValue - so some things will work, but GetUnderlyingType won't propagate anything
+									// It's special cased to do that.
+									AddReturnValue (value);
+								}
 							}
+							// We want to skip adding the `value` to the return Value because we have already added Nullable<value>
+							continue;
 						}
 						// We haven't found any generic parameters with annotations, so there's nothing to validate.
+					} else if (value == NullValue.Instance) {
+						// Do nothing - null value is valid and should not cause warnings nor marking
```

Comment:
You're right. The end result is the same since any reflection API will ignore the null on input anyway, but it's better to track it as empty here.
And the comment is wrong. MakeGenericType throws if "this" is null - obviously :-)